src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java

```diff
@@ -188,9 +195,51 @@ public List<String> validateHearingNumber(CaseData caseData, CorrespondenceType
         return errors;
     }
 
-    public void validateJurisdictionCodes(CaseData caseData, List<String> errors) {
+    public void validateJurisdiction(CaseData caseData, List<String> errors) {
         validateDuplicatedJurisdictionCodes(caseData, errors);
         validateJurisdictionCodesExistenceInJudgement(caseData, errors);
+        validateDisposalDate(caseData, errors);
+    }
+
+    private void validateDisposalDate(CaseData caseData, List<String> errors) {
+        if (CollectionUtils.isNotEmpty(caseData.getJurCodesCollection())) {
+            for (JurCodesTypeItem jurCodesTypeItem : caseData.getJurCodesCollection()) {
+                String disposalDate = jurCodesTypeItem.getValue().getDisposalDate();
+                if (!Strings.isNullOrEmpty(disposalDate)) {
+                    addInvalidDisposalDateError(caseData.getHearingCollection(), disposalDate, errors);
+                }
+            }
+        }
+    }
+
+    private void addInvalidDisposalDateError(List<HearingTypeItem> hearingTypeItems, String disposalDate, List<String> errors) {
+
+        LocalDate d = LocalDate.parse(disposalDate);
+        LocalDateTime dt = d.atTime(0,0, 0, 0);
+        if (HearingsHelper.isDateInFuture(
+                dt.toString(), LocalDateTime.now())
+        ) {
+            errors.add(DISPOSAL_DATE_IN_FUTURE);
+            return;
+        }
+        if (CollectionUtils.isNotEmpty(hearingTypeItems)) {
+            for (HearingTypeItem hearingTypeItem : hearingTypeItems) {
+                if (CollectionUtils.isEmpty(hearingTypeItem.getValue().getHearingDateCollection())) {
+                    continue;
+                }
+                for (DateListedTypeItem dateListedTypeItem : hearingTypeItem.getValue().getHearingDateCollection()) {
+                    if (areDatesEqual(disposalDate, dateListedTypeItem.getValue().getListedDate())) {
+                        return;
+                    }
+                }
+            }
+        }
+        errors.add(DISPOSAL_DATE_HEARING_DATE_MATCH);
+
+    }
+
+    private boolean areDatesEqual(String disposalDate, String hearingDate)  {
+        return disposalDate.split("T")[0].equals(hearingDate.split("T")[0]);
```

Comment:
I'm a bit nervous this code is not defensive and relying upon string comparisons for dates is a bad idea too.
Prefer to convert to dates and compare as dates instead, but also verify no nulls are involved too.

Reply:
dates are compared now